topostats/utils.py

```diff
@@ -76,28 +76,30 @@ def find_images(base_dir: Union[str, Path] = None, file_ext: str = ".spm") -> Li
     return list(base_dir.glob("**/*" + file_ext))
 
 
-def get_out_path(
-    image_path: Union[str, Path] = None, base_dir: Union[str, Path] = None, output_dir: Union[str, Path] = None
-) -> Path:
+def get_out_path(image_path: Path = None, base_dir: Path = None, output_dir: Path = None) -> Path:
     """Replaces the base directory part of the image path with the output directory.
 
     Parameters
     ----------
-    image_path: Union[str, Path]
+    image_path: Path
         The path of the current image.
-    base_dir: Union[str, Path]
-        Directory to recursively search for files, if not specified the current directory is scanned.
-    output_dir: Union[str, Path]
+    base_dir: Path
+        Directory to recursively search for files.
+    output_dir: Path
         The output directory specified in the configuration file.
 
     Returns
     -------
     Path
         The output path that mirrors the input path structure.
     """
-    pathparts = list(image_path.parts)
-    inparts = list(base_dir.parts)
-    return Path(output_dir / Path(*pathparts[len(inparts) :]))
+    try:
```

Comment:
I suspect I'll get a merge conflict when I come to merge #338 but I'll resolve that.

Reply:
Seems like your fix for it works pretty well too!